modules/src/handler.rs

```diff
@@ -1,5 +1,8 @@
 use crate::events::IbcEvent;
-use std::marker::PhantomData;
+use alloc::string::String;
+use core::marker::PhantomData;
```

Comment:
Why is this needed? Reading the docs looks like an advanced feature (FFI.. lifetimes which we barely have any).

I'm concerned that the codebase is drifting in a way that developers (speaking for myself mostly) can barely keep up. What I have in mind is the change to macros for handling errors, generics, and now this. The question is: Should the developer care about _anything_, when it comes to changes introduced by this PR, beside paying attention to avoid importing from `std`? If no, then good, that's simple. If yes (such as using or not using `PhantomData`), then @soareschen it would be good to:
1. spread awareness in the weekly call and make sure everyone understands the implications
2. start from the assumption that the codebase should remain as plain as possible. if we can avoid this, let's avoid it.

Reply:
Do you mean why is `core::marker::PhantomData` needed? I'm not sure if it is related to this PR, as it was already being used as `std::marker::PhantomData`. But I can explain the use of `PhantomData`.

The need for `PhantomData` is pretty specific for Rust, in that Rust requires all type parameters to be used. So if you have a struct that has unused type parameters for various practical reasons, you add a dummy field with the `PhantomData` type, which has zero runtime cost. The other more advanced use of `PhantomData` like you described are less common, and there is usually no need to concern about that.

The main change in this PR is to import constructs from `core` and `alloc` crates rather than `std`. But it seems like this is bringing a lot of confusion, even when it is only a name change with no technical details of its own. I guess this is mainly also because there are very little articles that explain the existence of the `core` and `alloc` crates.

I guess a good compromise is to follow @romac's suggestion to have our own surrogate `std` crate that re-exports everything from `core::prelude` and `alloc::prelude`. I am still worried that thing will bring unnecessary confusion to developers when they cannot find constructs like `std::thread` from this "fake" `std` crate. But maybe the benefit of being able to continue using `std` outweights that.

@romac what do you think?

Reply:
> The need for PhantomData is pretty specific for Rust...

Thank you for the explanation, that helps. If I understood correctly, here we have a `HandlerOutputBuilder` type that is generic over some type `T`. The problem, however, is that `T` does not appear as any field in `HandlerOutputBuilder`, and for that reason we need to add `T` as a fake "phantom" field, and the way to do that is as a `marker: PhantomData<T>` field in this type.

> Do you mean why is core::marker::PhantomData needed? I'm not sure if it is related to this PR, as it was already being used as std::marker::PhantomData.

You're right. The change from this PR surprised me as it highlighted the use of `PhantomData` which I don't understand. I wrongly assumed we introduced this construct with this PR. I wanted to make sure we're not introducing new constructs unless really necessary, as it may be difficult to maintain it without devs understanding them. But this is non-problem, please ignore my rambling here then! The two ideas I wrote above still stand though, when anyone makes important changes to the codebase they should:

> 1. spread awareness in the weekly call and make sure everyone understands the implications
> 2. start from the assumption that the codebase should remain as plain as possible. if we can avoid something, let's avoid it.

In this particular case, if we change from `std::...` to `core::...` it would be good to broadcast it to everyone and talk about it in the weekly call (I will add to agenda).